### PR TITLE
	fix issue about replication rule delete and username autocomplete

### PR DESCRIPTION
**Pull request by pengpengshui**

### CODE REVIEW

**Comment by reasonerjt** [comment]
Let's squash it before merging.

File: src/ui_ng/lib/src/replication/replication.component.ts [hunk 1]
```diff
@@ -106,6 +106,22 @@ export class ReplicationComponent implements OnInit {
   toggleJobSearchOption = optionalSearch;
   currentJobSearchOption: number;
 
+  public get canDeleteRule(): boolean  {
```
Comment (steven-zou) [inline src/ui_ng/lib/src/replication/replication.component.ts]: Should do the justify based on all the jobs

File: src/ui_ng/lib/src/replication/replication.component.ts [hunk 2]
```diff
@@ -106,6 +106,22 @@ export class ReplicationComponent implements OnInit {
...
+    let count: number = 0;
+    if (this.jobs && this.jobs.length) {
+      this.jobs.filter((job) => {
+        if ((job.status === 'pending') || (job.status === 'running') || (job.status === 'retrying')) {
+          count = count + 1;
+        }
+      });
+      if (count > 0) {
+        return false;
+      }
```
Comment (steven-zou) [inline src/ui_ng/lib/src/replication/replication.component.ts]: Simplify codes

File: src/ui_ng/lib/src/replication/replication.component.ts [hunk 3]
```diff
@@ -197,12 +213,19 @@ export class ReplicationComponent implements OnInit {
         this.totalCount = response.metadata.xTotalCount;
         this.jobs = response.data;
 
+        for(let i = 0; i< this.jobs.length; i++) {
```
Comment (steven-zou) [inline src/ui_ng/lib/src/replication/replication.component.ts]: Use subscription.timer()

File: src/ui_ng/src/app/project/member/add-member/add-member.component.ts [hunk 4]
```diff
@@ -97,6 +101,21 @@ export class AddMemberComponent implements AfterViewChecked, OnInit, OnDestroy {
               .catch(error => {
                 this.checkOnGoing = false;
               });
+            //username autocomplete
+            this.userService.getUsers()
```
Comment (steven-zou) [inline src/ui_ng/src/app/project/member/add-member/add-member.component.ts]: Use backend api to query user

File: src/ui_ng/lib/src/replication/replication.component.ts [hunk 5]
```diff
@@ -197,12 +207,30 @@ export class ReplicationComponent implements OnInit {
         this.totalCount = response.metadata.xTotalCount;
         this.jobs = response.data;
 
+        if (!this.timerDelay) {
+          this.timerDelay = Observable.timer(5000, 10000).subscribe(() => {
```
Comment (steven-zou) [inline src/ui_ng/lib/src/replication/replication.component.ts]: 5000=>10000

File: src/ui_ng/src/app/project/member/add-member/add-member.component.ts [hunk 6]
```diff
@@ -97,6 +108,20 @@ export class AddMemberComponent implements AfterViewChecked, OnInit, OnDestroy {
               .catch(error => {
                 this.checkOnGoing = false;
               });
+            //username autocomplete
+            if (this.userLists.length) {
+              this.selectUserName.length = 0;
```
Comment (steven-zou) [inline src/ui_ng/src/app/project/member/add-member/add-member.component.ts]: this.selectUserName=[]

File: src/ui_ng/src/app/project/member/add-member/add-member.component.ts [hunk 7]
```diff
@@ -69,11 +72,19 @@ export class AddMemberComponent implements AfterViewChecked, OnInit, OnDestroy {
...
+  selectUserName: string[] = [];
+  userLists: User[];
 
   constructor(private memberService: MemberService,
+    private userService: UserService,
     private messageHandlerService: MessageHandlerService,
     private translateService: TranslateService,
-    private ref: ChangeDetectorRef) { }
+    private ref: ChangeDetectorRef) {
+      this.userService.getUsers()
```
Comment (steven-zou) [inline src/ui_ng/src/app/project/member/add-member/add-member.component.ts]: Move out of constructor and move to ngOnInit

**Pull request closed without merging**